### PR TITLE
23272 rdbms support user tasks 2

### PR DESCRIPTION
**Pull request by p-wunderlich**
## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

#23272

### CODE REVIEW

File: zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/UserTaskExportHandler.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  private final UserTaskWriter userTaskWriter;
+
+  public UserTaskExportHandler(final UserTaskWriter userTaskWriter) {
+    this.userTaskWriter = userTaskWriter;
+  }
+
+  @Override
+  public boolean canExport(final Record<UserTaskRecordValue> record) {
+    return record.getIntent() == UserTaskIntent.CREATED
```
Comment (stoerti) [inline zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/UserTaskExportHandler.java:31]: 🔧  refactor to Set#contains() on a static Set?

File: db/rdbms/src/main/resources/db/changelog/rdbms-support/changesets/create_engine.xml [hunk 2]
```diff
@@ -103,5 +103,45 @@
...
+      <column name="DUE_DATE" type="TIMESTAMP WITH TIME ZONE(3)"/>
+      <column name="FOLLOW_UP_DATE" type="TIMESTAMP WITH TIME ZONE(3)"/>
+      <column name="EXTERNAL_FORM_REFERENCE" type="VARCHAR(200)"/>
+      <column name="PROCESS_DEFINITION_VERSION" type="SMALLINT"/>
+      <column name="CUSTOM_HEADERS" type="CLOB"/>
+      <column name="PRIORITY" type="INT"/>
+    </createTable>
+
+    <createTable tableName="CANDIDATE_USER">
+      <column name="USER_TASK_KEY" type="BIGINT"/>
```
[the remark this answers is not in the document]
Comment (p-wunderlich) [reply db/rdbms/src/main/resources/db/changelog/rdbms-support/changesets/create_engine.xml:131]: I added both

File: db/rdbms/src/main/resources/db/changelog/rdbms-support/changesets/create_engine.xml [hunk 3]
```diff
@@ -103,5 +103,45 @@
...
+      <column name="CANDIDATE_USER" type="VARCHAR(200)"/>
+    </createTable>
+
+    <createTable tableName="CANDIDATE_GROUP">
+      <column name="USER_TASK_KEY" type="BIGINT"/>
```
[the remark this answers is not in the document]
Comment (p-wunderlich) [reply db/rdbms/src/main/resources/db/changelog/rdbms-support/changesets/create_engine.xml:136]: Added both

File: db/rdbms/src/main/resources/mapper/UserTaskMapper.xml [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    statementType="PREPARED">
+    INSERT INTO CANDIDATE_GROUP (USER_TASK_KEY, CANDIDATE_GROUP)
+    VALUES
+    <foreach collection="candidateGroups" item="candidateGroup" separator=", ">
+      (#{key}, #{candidateGroup})
+    </foreach>
+  </insert>
+
+  <resultMap id="searchResultMap" type="io.camunda.db.rdbms.write.domain.UserTaskDbModel">
+    <constructor>
```
[the remark this answers is not in the document]
Comment (p-wunderlich) [reply db/rdbms/src/main/resources/mapper/UserTaskMapper.xml:56]: Good idea, changed it.

File: zeebe/exporters/rdbms-exporter/src/test/java/io/camunda/exporter/rdbms/DateUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import org.junit.jupiter.api.Test;
+
+public class DateUtilTest {
+
+  @Test
+  public void testToOffsetDateTimeFromLong() {
+    // Arrange
```
[the remark this answers is not in the document]
Comment (p-wunderlich) [reply zeebe/exporters/rdbms-exporter/src/test/java/io/camunda/exporter/rdbms/DateUtilTest.java:21]: O, copilot did weird stuff there. :D Bad Guy! I told him to always use given when then!

**Review by stoerti — APPROVED** [review]

**Pull request merged**